### PR TITLE
Ci to run tests and lint the code

### PR DESCRIPTION
**Pull request by mirzakhany**

### CODE REVIEW

**Comment by ccoVeille** [comment]
I had the issue locally earlier today. I'm unsure why the tests cannot be run without installing a lot of things.

Maybe it's normal, I didn't check further.

Maybe the image to used shouldn't be Ubuntu/alpine latest but something that will come with everything needed with gio.

Or mayve there are things in the code that could/should be removed to allow the test to be launched without much dependencies

**Comment by mirzakhany** [comment]
@ccoVeille Its because of GIo library dependency on C. 
Im trying to fix something similar to this here https://github.com/crypto-power/cryptopower/blob/master/.github/workflows/go.yml

**Comment by ccoVeille** [comment]
Except if the code does check on local lib in init methods. I don't see how your current tests could fail when launched.

The solution you mention may work but it looks strange to me to install this on each run. There should have gio ready docker image to use directly.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          only-new-issues: true
+          install-mode: "binary"
+
+      - name: Libraries for gio
+        run: |
+          sudo apt update
+          sudo apt install libwayland-dev libx11-dev libx11-xcb-dev libxkbcommon-x11-dev libgles2-mesa-dev libegl1-mesa-dev libffi-dev libxcursor-dev libvulkan-dev
+
+      - name: Lint Test
+        run: go test -v ./... && golangci-lint run --config=golangci.yaml --issues-exit-code=0
```
[the remark this answers is not in the document]
Comment (ccoVeille) [reply .github/workflows/ci.yaml:39]: Also you are already launching golangci-lint line 26

File: golangci.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+linters:
```
Comment (ccoVeille) [inline golangci.yaml:1]: Please rename this file to `.golangci-lint.yaml`

Then you won't have to use --config parameter

File: golangci.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  enable-all: false
```
Comment (ccoVeille) [inline golangci.yaml:2]: If you disable all, and disable a list, turn you simply have nothing enabled

**Review by ccoVeille — CHANGES_REQUESTED** [review]

**Comment by ccoVeille** [comment]
Anyway, I can help with golangci-lint.

**Pull request closed without merging**